### ISSUE TITLE
Bugs about Tooltip

### ISSUE THREAD

**Issue by ywu151** [uber-web/baseui#288]
The onClick function of its child component will become useless.
The tooltip will not disappear even when losing cursor after mouse moves away if mouse moves very fast.

* [ ] I have searched the [issues](https://github.com/uber-web/baseui/issues) of this repository and believe that this is not a duplicate.

## Expected Behavior

1. The `onClick` function of its child component will become useless. Such as in this case:
```
      <StatefulTooltip
        content={`Last failure: ${timestampToISOString(last_failure * 1000)}`}
        placement={PLACEMENT.bottom}
        key={key}
      >
        {itemContent}
      </StatefulTooltip>
```
```
    const itemContent = 
      <ItemWrapper
        status={status}
        key={key}
        numberitem={numberItem}
        onClick={partial(handleSearchSingleItem, {type, name})}
      >
        {name}
      </ItemWrapper>
```

2. If mouse moves very fast, the tooltip will not disappear even when losing cursor after mouse moves away.
![image](https://user-images.githubusercontent.com/33992458/45852400-80939d80-bcf4-11e8-8643-cba7e3e95514.png)
 

## Your Environment

| Tech    | Version |
| ------- | ------- |
| Base UI | v1.10.0  |
| React   |         |
| browser |         |
| node.js |         |
| etc.    |         |

## Steps to Reproduce

Link:

1.

2.

3.

4.

**Comment by schnerd** [comment]
First issue is fixed by #290

Second issue I'm going to dig into a bit further.

**Comment by gergelyke** [comment]
Released in 1.11.0 🚢
